slides/05-feature-engineering.qmd

```diff
@@ -139,19 +139,43 @@ tidymodels_prefer()
 str(season_2015)
 ```
 
+## Data splitting strategy
+
+```{r spending-diagram, echo = FALSE}
+#| fig-align: "center"
+
+knitr::include_graphics("images/whole-game-split.svg")
+```
+
+
 ## Splitting the NHL data `r hexes("rsample")`
 
 ```{r split}
 set.seed(23)
 nhl_split <- initial_split(season_2015, prop = 3/4)
 nhl_split
 
-nhl_train <- training(nhl_split)
+nhl_train_and_val <- training(nhl_split)
 nhl_test  <- testing(nhl_split)
 
-c(training = nrow(nhl_train), testing = nrow(nhl_test))
+c(`not testing` = nrow(nhl_train_and_val), testing = nrow(nhl_test))
+```
+
+## Validation split `r hexes("rsample")`
+
+Since there are a lot of observations, we'll use a validation set: 
```

Comment:
Can you say more about why we use a validation set over cross validation? i.e. they will have just spent some time working with cross validation, why switch to validation sets now?

What does "a lot of observations" have to do with it?

Even if these answers dont make it on the slide, they would make good speaker notes

Reply:
I added a slide. 

Basically, when you have enough data, doing multiple resamples take forever. Also, since the data are "large", the CV estimates are more precise that what is probably needed. 

A single resample can be precise enough with a good amount of data. 

Plus, since people use these a fair amount, we wanted to demonstrate it here.